File: TypeScript/URL/get-username-from-url.ts
import * as url from "url"

function main(): void {
    // Get the username from a given url.
    console.log(getUsernameFromURL("https://user:pass@www.example.com"))
}

main()

// Get the username from a given url.
function getUsernameFromURL(givenURL: string): string | null {
    return new URL(givenURL).username
}

File: Python/URL/get_username_from_url.py
import urllib.parse


# Get the username from a given url.
def get_username_from_url(url):
    return urllib.parse.urlparse(url).username


def main():
    # Get the username from a given url.
    print(get_username_from_url("https://user:pass@example.com:80"))


main()
